h2o-algos/src/main/java/hex/gam/GAM.java

```diff
@@ -107,16 +107,24 @@ public GAM(GAMModel.GAMParameters parms, Key<GAMModel> key) {
   // this function will check and make sure the knots location specified in knots are valid in the following sense:
   // 1. They do not contain NaN
   // 2. They are sorted in ascending order.
-  public void failVerifyKnots(double[] knots) {
+  public void failVerifyKnots(double[] knots, int gam_column_index) {
     for (int index = 0; index < knots.length; index++) {
       if (Double.isNaN(knots[index])) {
         error("knots formation", "default knots generation contain NaN.   Please specify it" +
                 " manually");
         return;
       }
-      if (index > 0 && knots[index - 1] >= knots[index]) { // knots must be increasing in order
-        error("knots formation", String.format("knots not sorted in ascending order. Knots at index %d: %f." +
-                "  Knots at index %d: %f", index-1, knots[index-1], index, knots[index]));
+      if (index > 0 && knots[index - 1] > knots[index]) {
+        error("knots formation", String.format("knots not sorted in ascending order for gam_column %s. " +
```

Comment:
the first argument should be the name of the field the error pertains to, "knots formation" doesn't seem to be a name of a field

Reply:
In this case, I am trying to specify what action is causing the problem.  In this case, it is in the knots formation.  I did specify which gam_column this is for in the later message.  What do you think?

Reply:
If this is a convention in H2O (I know we do use this across algo validations, not sure if everywhere, don't have precise stats), I'd be all for keeping the convention, as users will be instantly familiar with what is going on and which field to adjust.